packages/grafana-data/src/dataframe/StreamingDataFrame.test.ts

```diff
@@ -354,6 +354,95 @@ describe('Streaming JSON', () => {
     `);
   });
 
+  describe('keep track of packets', () => {
+    const json: DataFrameJSON = {
+      schema: {
+        fields: [
+          { name: 'time', type: FieldType.time },
+          { name: 'value', type: FieldType.number },
+        ],
+      },
+      data: {
+        values: [
+          [100, 200, 300],
+          [1, 2, 3],
+        ],
+      },
+    };
+
+    const stream = new StreamingDataFrame(json, {
+      maxLength: 4,
+      maxDelta: 300,
+    });
+
+    const getSnapshot = (f: StreamingDataFrame) => {
+      return {
+        values: f.fields[1].values.toArray(),
+        info: f.packetInfo,
+      };
+    };
+
+    expect(getSnapshot(stream)).toMatchInlineSnapshot(`
+      Object {
+        "info": Object {
```

Comment:
hmm, when used without the TS type, this naming and nested props are a bit confusing. lastPacketInfo, or lastPacketMeta. 

number may be better as sequence or seqNum or seqId.

Reply:
though action and sequential number are not really packet properties.

maybe leave as-is. we can tweak later. stream.info.length and stream.info.number dont jump out at me as info about the last action / packet but of the entire stream (which they aren't).